commands/create-account.js

```diff
@@ -48,10 +48,23 @@ async function createAccount(options) {
         keyPair = await KeyPair.fromRandom('ed25519');
         publicKey = keyPair.getPublicKey();
     }
-    await near.createAccount(options.accountId, publicKey);
     if (keyPair) {
         await near.connection.signer.keyStore.setKey(options.networkId, options.accountId, keyPair);
     }
+    try {
+        await near.createAccount(options.accountId, publicKey);
+    } catch(error) {
+        if (error.message.includes('Timeout')) {
+            console.warn('Received a timeout when creating account, please run:');
+            console.warn(`near state ${options.accountId}`);
+            console.warn('to confirm creation. Keyfile for this account has been saved.');
+        } else {
+            near.connection.signer.keyStore.removeKey(options.networkId, options.accountId)
+                .then(() => {
```

Comment:
@mikedotexe avoid mixing `await` and `.then` or you end up having unexpected results (eg. here error won't propagate properly to return value of this function.

Reply:
I don't understand why, but I tried using await and it didn't work.

Reply:
May you give more context what did not work? 

Current solution has some issues:
 - code doesn't wait until `removeKey` call completion
 - error gets ignored after logging (maybe that's intentional, but other errors usually would result in non-zero return code for shell)

It seems like it should work like this:
```js
await near.connection.signer.keyStore.removeKey(options.networkId, options.accountId);
throw error;
```

or if the intent is to swallow `removeKey` error like this:

```js
try {
    await near.connection.signer.keyStore.removeKey(options.networkId, options.accountId);
} catch(innerError) {
    console.error(`Error removing key for ${options.accountId}`, innerError);
} finally {
    throw error;
}

```


Let me know if you need help debugging this – let's figure it out together. Feel free to ping me next time if you have any questions regarding `await` / promises.

Reply:
I agree, let's fix this.

Reply:
@mikedotexe this is not fixed

Reply:
> [#402 (comment)](https://github.com/near/near-shell/pull/402#discussion_r437127111) looks like this not fixed yet @mikedotexe

@vgrichina we talked about this during our 1-on-1, remember? I shared my screen and showed you that `await` and `.then` behaved differently for some unknown reason.

Reply:
Now I remember that we talked about this, but I don't exactly remember what was the problem already – so good idea to add a comment. Looks like I have in the notes that key didn't get removed when using `.removeKey`, is this correct? I'll try to reproduce it and fix.

Even if we have to use `.then` there are other problems:
1) Looks like error won't propagate and process will return success code even though command failed.
2) Seems that `console.log(`Account ${options.accountId} for network "${options.networkId}" was created.`);` will print the message anyway, even though account wasn't created successfully.